shim/src/ctrl/getsockopt.c

```diff
@@ -58,5 +59,5 @@ int __getsockopt(int sockfd, int level, int optname, void *optval, socklen_t *op
         WARN("%s is not supported", "TCP_ULP");
     }
 
-    return (demi_getsockopt(sockfd, level, optname, optval, optlen));
+    return (__demi_getsockopt(sockfd, level, optname, optval, optlen));
```

Comment:
Shouldn't we do the same that we do in `getpeername()`? That isCheck if return value is different from zero. If it is, we should set errno and return `-1`.